### PR TITLE
Miscellaneous fixes for --disable-monitor mode

### PR DESCRIPTION
**Pull request by jchampio**
Here are the first few things I ran into while investigating https://github.com/citusdata/pg_auto_failover/pull/483. See commit messages for details.

### CODE REVIEW

File: src/bin/pg_autoctl/fsm_transition.c [hunk 1]
```diff
@@ -149,10 +149,10 @@ fsm_init_primary(Keeper *keeper)
 		/*
 		 * We have a new system_identifier, we need to publish it now.
 		 */
-		if (!monitor_set_node_system_identifier(
-				monitor,
-				keeper->state.current_node_id,
-				pgSetup->control.system_identifier))
+		if (!config->monitorDisabled &&
```
Comment (DimCitus) [inline src/bin/pg_autoctl/fsm_transition.c:152]: I'm on the fence to saying that I prefer having the monitorDisabled test in its own statement, even when everything is indented one level more, to make it very obvious when reading the code... it's nitpicking at this point though.

**Review by DimCitus — APPROVED** [review]
Looks good to me, thanks!

**Comment by DimCitus** [comment]
> About the Travis failure, I have another context in which I've seen it and I will prepare a separate PR that fixes the problem, I believe. Also we need more tests around synchronous_standby_names on the monitor and on the primary in the vein of what you've contributed already.

See https://github.com/citusdata/pg_auto_failover/pull/488

File: src/bin/pg_autoctl/fsm_transition.c (hunk 1 above)
Comment (jchampio) [reply src/bin/pg_autoctl/fsm_transition.c]: Done.

**Comment by jchampio** [comment]
> and maybe you had other edits you wanted to add to this PR before we merge it?

Nope, this is it for now.

> About the Travis failure, I have another context in which I've seen it and I will prepare a separate PR that fixes the problem, I believe.

The Travis failure is #481, I think.

**Comment by DimCitus** [comment]
See also https://github.com/citusdata/pg_auto_failover/pull/494.

**Comment by jchampio** [comment]
@DimCitus FYI, the squash commit that was pushed to `master` contains different authorship information for me than the original patches (personal vs work email). I think this is a [known issue](https://github.com/isaacs/github/issues/1368) when using some of GitHub's online tools.

**Pull request merged**